pypika/terms.py

```diff
@@ -336,24 +336,27 @@ def __init__(self, value: Any, alias: Optional[str] = None) -> None:
         self.value = value
 
     def get_value_sql(self, **kwargs: Any) -> str:
+        return self.get_formatted_value(self.value, **kwargs)
+
+    @classmethod
+    def get_formatted_value(cls, value: Any, **kwargs):
         quote_char = kwargs.get("secondary_quote_char") or ""
 
         # FIXME escape values
-        if isinstance(self.value, Term):
-            return self.value.get_sql(**kwargs)
-        if isinstance(self.value, Enum):
-            return self.value.value
-        if isinstance(self.value, date):
-            value = self.value.isoformat()
-            return format_quotes(value, quote_char)
-        if isinstance(self.value, str):
-            value = self.value.replace(quote_char, quote_char * 2)
+        if isinstance(value, Term):
+            return value.get_sql(**kwargs)
+        if isinstance(value, Enum):
+            return cls.get_formatted_value(value.value, **kwargs)
+        if isinstance(value, date):
+            return cls.get_formatted_value(value.isoformat(), **kwargs)
```

Comment:
Is there any need to call this method again here?  From what I can see it doesn't offer an improvement on the old date handling. Additionally, as `get_formatted_value` is called again, it causes value.replace to be called when it wasn't before.

Reply:
The idea was to convert `date` to one of the base types (`str` in this case) and then call whatever logic is implement for `str`. I think this is generally more DRY even if just slightly. So if you change the way you handle strings, you don't need to change it again for `date`, because `date` reuses the same logic.

Reply:
It might be overkill here, but I thought maybe other types are added later (just like Enum was now) and then we just apply the same logic.